pkg/processor/schemaregistry/client_test.go

```diff
@@ -0,0 +1,325 @@
+// Copyright © 2023 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schemaregistry
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"sync"
+	"testing"
+
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+	"github.com/conduitio/conduit/pkg/foundation/log"
+	"github.com/lovromazgon/franz-go/pkg/sr"
+	"github.com/matryer/is"
+)
+
+func TestClient_NotFound(t *testing.T) {
+	is := is.New(t)
+	ctx := context.Background()
+	logger := log.Nop()
+
+	rtr := newRoundTripRecorder(http.DefaultTransport)
+	c, err := NewClient(
+		logger,
+		sr.HTTPClient(&http.Client{Transport: rtr}),
+		sr.URLs(testSchemaRegistryURL(t)),
+	)
+	is.NoErr(err)
+
+	t.Run("SchemaByID", func(t *testing.T) {
+		is := is.New(t)
+
+		schema, err := c.SchemaByID(ctx, 12345)
+		is.True(err != nil)
+		is.Equal(sr.Schema{}, schema)
+
+		// check that error is expected
+		var respErr *sr.ResponseError
+		is.True(cerrors.As(err, &respErr))
+		is.Equal(40403, respErr.ErrorCode)
```

Comment:
Nitpick: it's maybe slightly more sefl-explanatory if we use that constant here, `errorCodeSchemaNotFound`.

Reply:
I realized this constant is in `client_fake_test.go` which is not compiled if we are executing an integration test. I left it as a magic number.

Honestly, I'd expect the client lib to provide these constants. I might work on opening a PR towards the lib we're using in my spare time.

Reply:
Agree!